public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -142,6 +143,22 @@ export function formValuesToRulerGrafanaRuleDTO(values: RuleFormValues): Postabl
   const { name, condition, noDataState, execErrState, evaluateFor, queries, isPaused, contactPoints, manualRouting } =
     values;
   if (condition) {
+    const notificationSettings: GrafanaNotificationSettings | undefined =
+      contactPoints?.grafana?.selectedContactPoint && manualRouting
+        ? {
+            receiver: contactPoints?.grafana?.selectedContactPoint,
+            mute_timings: contactPoints?.grafana?.muteTimeIntervals,
+            group_by: contactPoints?.grafana?.overrideGrouping ? contactPoints?.grafana?.groupBy : undefined,
+            group_wait: contactPoints?.grafana?.overrideTimings ? contactPoints?.grafana?.groupWaitValue : undefined,
+            group_interval: contactPoints?.grafana?.overrideTimings
+              ? contactPoints?.grafana?.groupIntervalValue
+              : undefined,
+            repeat_interval: contactPoints?.grafana?.overrideTimings
+              ? contactPoints?.grafana?.repeatIntervalValue
+              : undefined,
+          }
+        : undefined;
```

Comment:
Could we extract this fragment into a separate function where it will be easier to handle undefined values?

Reply:
good idea

Reply:
done!